### PR TITLE
build one nxp sample | remove snprintf telem payload in pnp

### PR DESCRIPTION
**Pull request by danewalton**
Temporary solution while we investigate building nxp files as a lib.

### CODE REVIEW

File: demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c [hunk 1]
```diff
@@ -865,8 +890,8 @@ static void prvAzureDemoTask( void * pvParameters )
         /* Publish messages with QoS1, send and process Keep alive messages. */
         for( ; ; )
         {
-            ulScratchBufferLength = snprintf( ( char * ) ucScratchBuffer, sizeof( ucScratchBuffer ),
-                                              sampleazureiotMESSAGE, xDeviceCurrentTemperature );
+            prvCreateTelemetryPayload(ucScratchBuffer, sizeof(ucScratchBuffer), &ulScratchBufferLength);
```
Comment (hihigupt) [inline demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c:893]: lets keep it simple with snprintf....To do that please add nano-spec, and uninitialize the printf/scanf.

File: demos/projects/NXP/mimxrt1060/gcc_flags.cmake [hunk 2]
```diff
@@ -30,6 +30,8 @@ set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} " \
 set(CMAKE_EXE_LINKER_FLAGS ${CMAKE_EXE_LINKER_FLAGS} " \
     -fno-exceptions \
     -fno-rtti \
```
Comment (hihigupt) [inline demos/projects/NXP/mimxrt1060/gcc_flags.cmake:32]: nano-spec is missing in both compile and linker steps. Can you also please make sure that map file shows the correct library is getting linked (and test...:))

Comment (danewalton) [reply demos/projects/NXP/mimxrt1060/gcc_flags.cmake:32]: yup lib nano is linked

Comment (danewalton) [reply demos/projects/NXP/mimxrt1060/gcc_flags.cmake:32]: image

**Review by wduraes — APPROVED** [review]

**Pull request merged**